test/source/tests/settings.ts

```diff
@@ -967,6 +969,88 @@ export const defineSettingsTests = (testVariant: TestVariant, testWithBrowser: T
+      };
+      mockBackendData.clientConfigurationByAcctEmail[acct] = setup;
+      const setupPage = await BrowserRecipe.openSettingsLoginApprove(t, browser, acct);
+      await SetupPageRecipe.autoSetupWithEKM(setupPage);
+      const {
+        cryptup_settingssettingsflowcrypttest_rules: rules1
+      } = await setupPage.getFromLocalStorage([
+        'cryptup_settingssettingsflowcrypttest_rules'
+      ]);
+      const clientConfiguration1 = rules1 as ClientConfiguration;
+      expect(clientConfiguration1.flags).to.eql([
+        'NO_PRV_BACKUP',
+        'ENFORCE_ATTESTER_SUBMIT',
+        'PRV_AUTOIMPORT_OR_AUTOGEN',
+        'PASS_PHRASE_QUIET_AUTOGEN',
+        'DEFAULT_REMEMBER_PASS_PHRASE']);
+      expect(clientConfiguration1.disallow_attester_search_for_domains).to.eql(['disallowed_domain1.test', 'disallowed_domain2.test']);
+      expect(clientConfiguration1.enforce_keygen_algo).to.equal('rsa2048');
+      expect(clientConfiguration1.key_manager_url).to.equal('https://localhost:8001/flowcrypt-email-key-manager');
+      await setupPage.close();
+
+      // modify the flags returned for the account
+      setup.flags = ['NO_ATTESTER_SUBMIT', 'HIDE_ARMOR_META', 'DEFAULT_REMEMBER_PASS_PHRASE'];
+      // modify other settings
+      setup.disallow_attester_search_for_domains = ['disallowed_domain3.test', 'disallowed_domain4.test'];
+      setup.enforce_keygen_algo = 'rsa3072';
+      // open the settings page
+      const settingsPage = await browser.newPage(t, TestUrls.extensionSettings(acct));
+      const {
+        cryptup_settingssettingsflowcrypttest_rules: rules2
+      } = await settingsPage.getFromLocalStorage(['cryptup_settingssettingsflowcrypttest_rules']);
+      // check that the configuration in the storage has been updated
+      const clientConfiguration2 = rules2 as ClientConfiguration;
+      expect(clientConfiguration2.flags).to.eql(['NO_ATTESTER_SUBMIT', 'HIDE_ARMOR_META', 'DEFAULT_REMEMBER_PASS_PHRASE']);
+      expect(clientConfiguration2.disallow_attester_search_for_domains).to.eql(['disallowed_domain3.test', 'disallowed_domain4.test']);
+      expect(clientConfiguration2.enforce_keygen_algo).to.equal('rsa3072');
+      const accessToken = await BrowserRecipe.getGoogleAccessToken(settingsPage, acct);
+      // keep settingsPage open to re-read the storage later via it
+      // re-configure the setup again
+      setup.flags = [
+        'NO_PRV_BACKUP',
+        'ENFORCE_ATTESTER_SUBMIT',
+        'PRV_AUTOIMPORT_OR_AUTOGEN',
+        'PASS_PHRASE_QUIET_AUTOGEN',
+        'DEFAULT_REMEMBER_PASS_PHRASE'];
+      setup.disallow_attester_search_for_domains = ['disallowed_domain5.test', 'disallowed_domain6.test'];
+      setup.enforce_keygen_algo = 'rsa2048';
+      const extraAuthHeaders = { Authorization: `Bearer ${accessToken}` };
+      const gmailPage = await browser.newPage(t, TestUrls.mockGmailUrl(), undefined, extraAuthHeaders);
```

Comment:
@tomholub These tests confirm that the updated `rules` get their way into the storage on every load of `settings` page or web content script. Is this sufficient? Should we worry about other possibly open tabs with older versions of `ClientConfiguration`?
Also, should we consider some scenarios when user interaction or warning is necessary?
For example, if `FORBID_STORING_PASS_PHRASE` rule has appeared and the passphrase is currently stored, should we clear it? What if it was previously generated in the silent mode?
Also, what if `enforce_keygen_algo` has changed? Should we make the user re-generate the keys?
Perhaps, you know much more about such possible scenarios.

Reply:
> @tomholub These tests confirm that the updated rules get their way into the storage on every load of settings page or web content script. Is this sufficient? Should we worry about other possibly open tabs with older versions of ClientConfiguration?

That's enough. Not worried about already open tabs elsewhere

> Also, should we consider some scenarios when user interaction or warning is necessary?
> For example, if FORBID_STORING_PASS_PHRASE rule has appeared and the passphrase is currently stored, should we clear it? What if it was previously generated in the silent mode?
> Also, what if enforce_keygen_algo has changed? Should we make the user re-generate the keys?
> Perhaps, you know much more about such possible scenarios.

We'll ignore such cases for now, but we'll file an issue. Once a customer actually needs to change a particular client configuration, we'll implement a mechanism for whatever they need to change. The most likely one to get implemented will probably be a chance of pass phrase storage, either from in-memory to stored or stored to in-memory.

Nobody is currently using auto-generated pass phrases in silent mode.